arbnode/sequencer.go

```diff
@@ -196,17 +211,28 @@ func (s *Sequencer) PublishTransaction(ctx context.Context, tx *types.Transactio
 	select {
 	case s.txQueue <- queueItem:
 	case <-ctx.Done():
-		return ctx.Err()
+		err := ctx.Err()
+		select {
+		case err = <-resultChan:
```

Comment:
I'm not sure why this is necessary, given that resultChan is only read below this which won't be hit. We also don't want to clog the result chan if the sequencer tries to post a result.

Reply:
:+1: